collect_app/src/main/java/org/odk/collect/android/widgets/CounterWidget.kt

```diff
@@ -0,0 +1,69 @@
+package org.odk.collect.android.widgets
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import org.javarosa.core.model.data.IAnswerData
+import org.javarosa.form.api.FormEntryPrompt
+import org.odk.collect.android.databinding.CounterWidgetBinding
+import org.odk.collect.android.formentry.questions.QuestionDetails
+import org.odk.collect.android.widgets.utilities.StringWidgetUtils
+
+class CounterWidget(
+    context: Context,
+    questionDetails: QuestionDetails
+) : QuestionWidget(context, questionDetails) {
+    lateinit var binding: CounterWidgetBinding
+
+    private var value: Int? = null
+        set(newValue) {
+            field = newValue?.takeIf { it in 0..999999999 }
```

Comment:
I've just realized I don't understand why this is `999999999` and not `Integer.MAX_VALUE` (which is `2147483647`).

Reply:
Because the standard integer widget allows up to 9 digits, the maximum supported value is 999,999,999 so I wanted to make it consistent.

Reply:
Ah, I didn't know that (but just saw it at https://docs.getodk.org/form-question-types/#number-widgets). Let's pull that out as a constant.

Reply:
Done.